src/server/repositories/UrlRepository.ts

```diff
@@ -521,7 +528,17 @@ export class UrlRepository implements UrlRepositoryInterface {
             )
             return
           }
-          resolve(cacheLongUrl)
+
+          try {
+            const redirectDestination = JSON.parse(cacheLongUrl)
```

Comment:
[issue] i think there are 2 issues here:

first, i think we updated the data structure from string -> json, which might cause issues when we're trying to read the new value. i think our new code cannot `reject` outright and we'd need to handle the fallback of parsing a string value rather than our json encoded value. 

secondly, because our deploy window is so long, we need to handle the case where our old code reads a json encoded value. this will cause our existing flow to mess up, so it needs to be handled properly too. 

from our offline discussion, i think we can do a small release so that we wrap the `JSON.parse` method here to try reading it as a JSON value first, then fallback to raw string